### PR TITLE
fix: swc parser should always preserve comments

### PR DESCRIPTION
**Pull request by h-a-n-a**
## Checklist

- [X] Tests updated (or not required).
- [X] Documentation updated (or not required).

### CODE REVIEW

File: packages/rspack-test-tools/tests/builtinCases/plugin-javascript/builtins-define/snapshot/output.snap.txt [hunk 1]
```diff
@@ -260,7 +261,8 @@ __webpack_require__.d(__webpack_exports__, {
 });
 const DO_NOT_CONVERTED7 = 402;
 const DO_NOT_CONVERTED9 = 403;
-/* harmony default export */ __webpack_exports__["default"] = (401); // DO_NOT_CONVERTED8
+/* harmony default export */ __webpack_exports__["default"] = (401);
+// DO_NOT_CONVERTED8
```
Comment (h-a-n-a) [inline packages/rspack-test-tools/tests/builtinCases/plugin-javascript/builtins-define/snapshot/output.snap.txt:265]: Changes to this file appear to be the same with swc playground.

File: packages/rspack-test-tools/tests/treeShakingCases/webpack-innergraph-try-globals/snapshot/new_treeshaking.snap.txt [hunk 2]
```diff
@@ -32,8 +32,9 @@ try {
     var c = b;
     const a = 42;
     var ok2 = false;
-    eval(""); // TODO terser has a bug and incorrectly remove this code, eval opts out
-} catch (e) {
+    eval("");
+} // TODO terser has a bug and incorrectly remove this code, eval opts out
```
Comment (h-a-n-a) [inline packages/rspack-test-tools/tests/treeShakingCases/webpack-innergraph-try-globals/snapshot/new_treeshaking.snap.txt:36]: Changes to this file appear to be the same with swc playground.

File: packages/rspack-test-tools/tests/treeShakingCases/tree-shaking-interop/snapshot/new_treeshaking.snap.txt [hunk 3]
```diff
@@ -30,8 +30,7 @@ exports.test = 30;
 {
     const res = __webpack_require__("./a.js");
     module.exports = res;
-} // export default function () {}
```
Comment (h-a-n-a) [inline packages/rspack-test-tools/tests/treeShakingCases/tree-shaking-interop/snapshot/new_treeshaking.snap.txt:33]: swc playground will keep this comment.

File: packages/rspack-test-tools/tests/treeShakingCases/webpack-innergraph-try-globals/snapshot/output.snap.txt [hunk 4]
```diff
@@ -33,8 +33,9 @@ try {
     var c = b;
     const a = 42;
     var ok2 = false;
-    eval(""); // TODO terser has a bug and incorrectly remove this code, eval opts out
-} catch (e) {
+    eval("");
+} // TODO terser has a bug and incorrectly remove this code, eval opts out
```
Comment (h-a-n-a) [inline packages/rspack-test-tools/tests/treeShakingCases/webpack-innergraph-try-globals/snapshot/output.snap.txt:37]: Changes to this file appear to be the same with swc playground.vvv

File: plugin-test/css-extract/enforceEsm.test.js [hunk 5]
```diff
@@ -65,8 +65,5 @@ it('should keep empty module when options.esModule is equal "false"', async () =
 	expect(
 		modules.filter(m => m.moduleType !== "runtime" && !m.orphan).length
 	).toBe(2);
-	expect(source("./simple.css", stats)).toMatchInlineSnapshot(`
-		"// extracted by css-extract-rspack-plugin
-		"
-	`);
+	expect(source("./simple.css", stats)).toMatchInlineSnapshot(`""`);
```
Comment (h-a-n-a) [inline plugin-test/css-extract/enforceEsm.test.js:68]: Changes to this file appear to be the same with swc playground.

**Review by LingyuCoder — APPROVED** [review]

**Pull request merged**